### PR TITLE
Add option to emit link to message instead of transcluding it

### PR DESCRIPTION
**Pull request by FND**
cf. #17

### CODE REVIEW

File: purpler/bot.py [hunk 1]
```diff
@@ -189,18 +189,31 @@ def on_pubmsg(self, c, e):
...
-                guid = result.group(1)
-                self.log.debug('saw guid %s', guid)
+                mode = result.group(1)
+                if mode == "l":
+                    link_only = True
+                elif mode == "t":
+                    link_only = False
+                else:
+                    # XXX silent failure
+                    continue
```
Comment (FND) [inline purpler/bot.py]: Is this what we want here?

The entire block seems a bit heavy-handed either way.

File: purpler/bot.py [hunk 2]
```diff
@@ -189,18 +189,31 @@ def on_pubmsg(self, c, e):
...
+                self.log.debug('saw guid %s (%s)', guid, mode)
                 outgoing_message = self.storage.get(guid)
                 if outgoing_message:
-                    out_nick, the_message = outgoing_message.content.split(':', 1)
-                    c.privmsg(e.target, '<%s> %s [%s] [n %s]' % (
-                        out_nick.strip(), the_message.strip(),
-                        outgoing_message.when,
-                        outgoing_message.guid))
+                    if link_only:
+                        msg = '%s/%s' % (self.web_url, outgoing_message.guid))
```
Comment (FND) [inline purpler/bot.py]: I'm not sure whether this is correct.

File: purpler/bot.py [hunk 3]
```diff
@@ -189,18 +189,31 @@ def on_pubmsg(self, c, e):
...
         if results:
             for result in results:
-                guid = result.group(1)
-                self.log.debug('saw guid %s', guid)
+                mode = result.group(1)
+                if mode == "l":
+                    link_only = True
+                elif mode == "t":
+                    link_only = False
+                else:
```
Comment (cdent) [inline purpler/bot.py]: is this even possible? group(1) can only be t or l according to the regex?

Comment (FND) [reply purpler/bot.py]: Uh, good point... :man_facepalming: Will rewrite.

**Comment by FND** [comment]
Not sure which query you're referring to.

I've simplified the code as per your comment.

**Comment by cdent** [comment]
> Not sure which query you're referring to

The one you fixed.

**Comment by FND** [comment]
You realize being cryptic doesn't equate mining bitcoin? I'm not aware of any such fix.

However, you made me spot spurious closing parentheses; fixed now.

**Comment by cdent** [comment]
Don't make me call jd in here to ask who that crazy obsessive person is.

**Comment by FND** [comment]
Can you assign a nid to this discussion? I'd like to have it chiseled into my headstone.

**Comment by cdent** [comment]
355388102

**Comment by FND** [comment]
I figured you hated the masons.

**Pull request merged**